evm_arithmetization/src/cpu/kernel/asm/hash/sha2/ops.asm

```diff
@@ -1,14 +1,14 @@
 // 32-bit right rotation
 %macro rotr(rot)
     // stack: value
+    DUP1
+    // stack: value, value
     PUSH $rot
-    // stack: rot, value
-    DUP2
-    DUP2
-    // stack: rot, value, rot, value
+    // stack: rot, value, value
     SHR
-    // stack: value >> rot, rot, value
-    %stack (shifted, rot, value) -> (rot, value, shifted)
+    // stack: value >> rot, value
+    SWAP1
+    PUSH $rot
```

Comment:
Yes I'm pushing against instead of calling `DUP` because `$rot` always fits in 4 bytes at most anyway, and the savings on the # CPU cycles are worth the extra overhead on `BytePackingStark`.